sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -749,6 +749,7 @@ class AdaptiveQueryExecSuite
     // so retry several times here to avoid unit test failure.
     eventually(timeout(15.seconds), interval(500.milliseconds)) {
       withSQLConf(
+        SQLConf.ARTIFACTS_SESSION_ISOLATION_ALWAYS_APPLY_CLASSLOADER.key -> "true",
```

Comment:
Hi, @xupefei and @HyukjinKwon . 

May I ask why this PR touches irrelevant `AdaptiveQueryExecSuite` here?

Reply:
Also, I'm observing this test case failure still in some environments although this PR claims that `These tests will not fail after this PR`.
> Also, some existing tests will fail when the config ARTIFACTS_SESSION_ISOLATION_ALWAYS_APPLY_CLASSLOADER is enabled:
> AdaptiveQueryExecSuite: SPARK-37753: Inhibit broadcast in left outer join when there are many empty partitions on outer/left side
> DatasetOptimizationSuite: SPARK-27871: Dataset encoder should benefit from codegen cache
> These tests will not fail after this PR.